### PR TITLE
_7zz: correct license

### LINKED ISSUE #163680: 7zz is licensed as unfree
### Describe the bug
The 7zip uses unrar's code. https://fedoraproject.org/wiki/Licensing:Unrar It's better to make the unrar part optional so that it's free by default.

### Steps To Reproduce
1. ...
2. ...
3. ...

### Notify maintainers

@peterhoeg @thiagokokada

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Description of changes

- `_7zz`: correct license and remove p7zip dependency
  - The code under `Compress/Rar*` is licensed under a specific unRAR license
  - Also `Compress/LzfseDecoder.cpp` is covered by BSD3
  - Meta has been updated and tweaked
  - Source is now downloaded from sourceforge in the `.tar.xz` version to avoid depending on `p7zip`
- `_7zz`: add notice of the license updates and optional unRAR licenced code

Resolves #163680

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [X] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/archivers/7zz/default.nix [hunk 1]
```diff
@@ -14,17 +24,25 @@ stdenv.mkDerivation rec {
...
 
   makefile = "../../cmpl_gcc${platformSuffix}.mak";
 
-  nativeBuildInputs = [ p7zip ] ++ lib.optionals useUasm [ uasm ];
+  nativeBuildInputs = lib.optionals useUasm [ uasm ];
+
+  # in addition to the make var, also remove Compress/Rar* code to be safe
+  postPatch = lib.optionalString (!enableUnfree) ''
+    rm -rf ../../Compress/Rar*
+  '';
```
Comment (06kellyjac) [inline pkgs/tools/archivers/7zz/default.nix:45]: this might be overkill but it shouldn't hurt :shrug:

**Comment by 06kellyjac** [comment]
> On top of that, the license requires that you agree to these use restrictions, or you must remove the software (source and binary) from your hard disks.

https://fedoraproject.org/wiki/Licensing:Unrar

We might actually want to strip the code from the src FOD :eyes:
Maybe that crap doesn't count if it's `.tar.xz`-ed up :shrug:

**Comment by peterhoeg** [comment]
From the link:

> Accordingly, this means that any code under this license can not be shipped as part of Fedora in source or binary form.

If you amend the source to fetch from here, it should be safe as I strip the Unrar sources:

https://gitlab.com/peterhoeg/7z-free

**Comment by 06kellyjac** [comment]
I'm not a huge fan of maintaining a repo following 7z without the unRAR code. The nix for grabbing the source and ripping out the unRAR code is a bit more complex but I've added an update script to handle that.

I should also probably sign up to maintain this mess I've made :sweat_smile:

**Comment by anna328p** [comment]
should we remove the unRAR code, or have an option in the package to enable it, and expose two packages? i use 7zz to extract RAR files very often.

**Comment by peterhoeg** [comment]
There is an option to enable it - it's just off by default.

**Comment by 06kellyjac** [comment]
exactly, you can either add an overlay to get it with unRAR globally:

```
final: prev: {
  _7zz = prev._7zz.override { enableUnfree = true; };
}
```

or just override it where you add it to your machine:

```
environment.systemPackages = with pkgs; [
  (_7zz.override { enableUnfree = true; })
];
```

**Comment by 06kellyjac** [comment]
update passthru was pointing at the wrong file. Pushed a fix

**Pull request merged**